pkg/subctl/cmd/show.go

```diff
@@ -74,22 +74,25 @@ func getMultipleRestConfigs(kubeConfigPath, kubeContext string) ([]restConfig, e
 	return restConfigs, nil
 }
 
-func getClientAndRawConfig(rules *clientcmd.ClientConfigLoadingRules, overrides *clientcmd.ConfigOverrides) (restConfig, error) {
+func getClientConfigAndClusterName(rules *clientcmd.ClientConfigLoadingRules, overrides *clientcmd.ConfigOverrides) (restConfig, error) {
 	config := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides)
 	clientConfig, err := config.ClientConfig()
 	if err != nil {
 		return restConfig{}, err
 	}
 
-	var context string
+	raw, err := config.RawConfig()
+	if err != nil {
+		return restConfig{}, err
+	}
+
+	var clusterName string
+
 	if overrides.CurrentContext != "" {
-		context = overrides.CurrentContext
+		clusterName = *getClusterNameFromContext(raw, overrides.CurrentContext)
 	} else {
-		raw, err := config.RawConfig()
-		if err != nil {
-			return restConfig{}, err
-		}
-		context = raw.CurrentContext
+		clusterName = *getClusterName(raw)
 	}
-	return restConfig{config: clientConfig, context: context}, nil
+
+	return restConfig{config: clientConfig, context: clusterName}, nil
```

Comment:
This ends up being somewhat confusing — `context` no longer contains the context, but the cluster name. I see that this fixes the bug in all `show` commands without causing any issues, but it would be better to rename the field.